includes/class-wc-stripe-database-cache.php

```diff
@@ -284,13 +323,14 @@ private static function maybe_trigger_prefetch( string $key, array $cache_conten
 	 * Adds the CACHE_KEY_PREFIX + plugin mode prefix to the key.
 	 * Ex: "wcstripe_cache_[mode]_[key].
 	 *
-	 * @param string $key The key to add the prefix to.
+	 * @param string $key       The key to add the prefix to.
+	 * @param string|null $mode The mode to use as prefix for the key. Default is null, which means the current plugin mode.
 	 *
 	 * @return string The key with the prefix.
 	 */
-	private static function add_key_prefix( $key ) {
-		$mode = WC_Stripe_Mode::is_test() ? 'test_' : 'live_';
-		return self::CACHE_KEY_PREFIX . $mode . $key;
+	private static function add_key_prefix( string $key, ?string $mode = null ): string {
+		$mode = $mode ?? ( WC_Stripe_Mode::is_test() ? 'test' : 'live' );
```

Comment:
As things stand, any string value for `$mode` will be accepted. Might it make sense to validate that we have `'live'`, `'test'`, or `null` at this stage?
```suggestion
		if ( null === $mode ) {
			$mode = WC_Stripe_Mode::is_test() ? 'test' : 'live';
		} elseif ( 'live' !== $mode && 'test' !== $mode ) {
			// Don't allow other values for $mode
			$mode = 'test';
		}
		// Otherwise $mode is either 'live' or 'test'
```